### PR TITLE
Changeling revisions

### PR DESCRIPTION
**Pull request by icantthinkofanameritenow**
Modifies some changeling power costs and adds the Arm Blade power by popular demand.I'm planning that this will encourage more aggressive play for lings and easier use of horror form. 

-Horror form cost is halved.
-Death Sting is now 8 points, and injects 20 units of Cyanide instead of 40 units of Lexorin. This will make it more obviously lethal, but since cyanide appears rarely an advanced mass-spectrometer will make it easy to deduce the cause of death as a changeling sting. 
-DNA extract sting is now 3 points.

### CODE REVIEW

**Comment by ghost** [comment]
On what logic are you basing these balance changes to changeling?

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 1]
```diff
@@ -937,7 +937,7 @@ var/list/datum/dna/hivemind_bank = list()
 	T.silent = 10
 	T.Paralyse(10)
 	T.Jitter(1000)
-	if(T.reagents)	T.reagents.add_reagent("lexorin", 40)
+	if(T.reagents)	T.reagents.add_reagent("cyanide", 20)
```
Comment (9600bauds) [inline code/game/gamemodes/changeling/changeling_powers.dm:940]: Worth noting this change in the OP

**Comment by ghost** [comment]
May want to add a changelog for a change like this.

**Comment by ghost** [comment]
Good changes, horror form is shit anyway. Maybe now it will see some use beyond 45 min monkey man farming.

**Comment by ghost** [comment]
Oh, tag ninja'd.

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 2]
```diff
@@ -962,3 +962,71 @@ var/list/datum/dna/hivemind_bank = list()
...
+	weapon_type = /obj/item/weapon/melee/arm_blade
+	weapon_name_simple = "blade"
+
+/obj/item/weapon/melee/arm_blade
+	name = "arm blade"
+	desc = "A grotesque blade made out of bone and flesh that cleaves through people as a hot knife through butter"
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "arm_blade"
+	item_state = "arm_blade"
+	flags = ABSTRACT | NODROP
```
Comment (ghost) [inline code/game/gamemodes/changeling/changeling_powers.dm]: I'm not sure we have a NODROP flag, I was sure we used a different variable for this.

Comment (icantthinkofanameritenow) [inline code/game/gamemodes/changeling/changeling_powers.dm]: We probably do, I just don't know it.

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 3]
```diff
@@ -962,3 +962,71 @@ var/list/datum/dna/hivemind_bank = list()
...
+	req_human = 1
+	max_genetic_damage = 20
+	weapon_type = /obj/item/weapon/melee/arm_blade
+	weapon_name_simple = "blade"
+
+/obj/item/weapon/melee/arm_blade
+	name = "arm blade"
+	desc = "A grotesque blade made out of bone and flesh that cleaves through people as a hot knife through butter"
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "arm_blade"
```
Comment (ghost) [inline code/game/gamemodes/changeling/changeling_powers.dm]: Gotta port the icons as well

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 4]
```diff
@@ -962,3 +962,71 @@ var/list/datum/dna/hivemind_bank = list()
...
+	flags = ABSTRACT | NODROP
+	w_class = 5.0
+	force = 25
+	throwforce = 0 //Just to be on the safe side
+	throw_range = 0
+	throw_speed = 0
+	sharpness = IS_SHARP
+
+/obj/item/weapon/melee/arm_blade/New(location,silent)
+	..()
```
Comment (ghost) [inline code/game/gamemodes/changeling/changeling_powers.dm]: Should there not be a check for whether or not the mob is a monkey somewhere around here?

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 5]
```diff
@@ -962,3 +962,71 @@ var/list/datum/dna/hivemind_bank = list()
...
+		var/obj/machinery/door/airlock/A = target
+
+		if(!A.requiresID() || A.allowed(user)) //This is to prevent stupid shit like hitting a door with an arm blade, the door opening because you have acces and still getting a "the airlocks motors resist our efforts to force it" message.
+			return
+
+		if(A.hasPower())
+			if(A.locked)
+				user << "<span class='warning'>The airlock's bolts prevent it from being forced!</span>"
+				return
+			user << "<span class='warning'>The airlock's motors are resisting, this may take time...</span>"
```
Comment (ghost) [inline code/game/gamemodes/changeling/changeling_powers.dm]: To_chat

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 6]
```diff
@@ -962,3 +962,71 @@ var/list/datum/dna/hivemind_bank = list()
...
+
+	else if(istype(target, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A = target
+
+		if(!A.requiresID() || A.allowed(user)) //This is to prevent stupid shit like hitting a door with an arm blade, the door opening because you have acces and still getting a "the airlocks motors resist our efforts to force it" message.
+			return
+
+		if(A.hasPower())
+			if(A.locked)
+				user << "<span class='warning'>The airlock's bolts prevent it from being forced!</span>"
```
Comment (ghost) [inline code/game/gamemodes/changeling/changeling_powers.dm]: to_chat(user, message)

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 7]
```diff
@@ -962,3 +962,71 @@ var/list/datum/dna/hivemind_bank = list()
...
+				return
+			user << "<span class='warning'>The airlock's motors are resisting, this may take time...</span>"
+			if(do_after(user, 100, target = A))
+				A.open(2)
+			return
+
+		else if(A.locked)
+			user << "<span class='warning'>The airlock's bolts prevent it from being forced!</span>"
```
Comment (ghost) [inline code/game/gamemodes/changeling/changeling_powers.dm]: To chat again

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 8]
```diff
@@ -962,3 +962,71 @@ var/list/datum/dna/hivemind_bank = list()
...
+
+/obj/item/weapon/melee/arm_blade/dropped(mob/user)
+	qdel(src)
+
+/obj/item/weapon/melee/arm_blade/afterattack(atom/target, mob/user, proximity)
+	if(!proximity)
+		return
+	if(istype(target, /obj/structure/table))
+		var/obj/structure/table/T = target
+		T.table_destroy(1)
```
Comment (ghost) [inline code/game/gamemodes/changeling/changeling_powers.dm]: Also works on reinforced tables. Intentional?

File: code/game/gamemodes/changeling/changeling_powers.dm (hunk 4 above)
Comment (ghost) [inline code/game/gamemodes/changeling/changeling_powers.dm]: Whoops, there's a check for it somewhere else

Comment (icantthinkofanameritenow) [inline code/game/gamemodes/changeling/changeling_powers.dm]: Yeah, there should be. Pardon the messy work, this is my first attempt porting anything and don't know how to do it very well. I'll try to port the parent of the armblade code in, but I'm not sure if that'll work right either.

**Comment by SangriaMonkey** [comment]
:ok_hand:

**Comment by ghost** [comment]
> -Adds the Fake Arm Blade Sting,

That is terrible and should never be possible to so easily frame other of being a ling or to disable ones hand.

**Comment by Shadowmech88** [comment]
Another affect of the death sting change is that now people can't carry around 50u dexalin pills as a death sting countermeasure.

**Comment by NigglyWiggly** [comment]
Did people carry around 50u Dex pills before?

**Comment by 9600bauds** [comment]
No, oxyloss first aid kits already had them for you.

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 9]
```diff
@@ -962,3 +962,91 @@ var/list/datum/dna/hivemind_bank = list()
...
+		user.visible_message("<span class='warning'>With a sickening crunch, [user] reforms their [weapon_name_simple] into an arm!</span>", "<span class='notice'>We assimilate the [weapon_name_simple] back into our body.</span>", "<span class='italics>You hear organic matter ripping and tearing!</span>")
+		if(right_hand)
+			user.update_inv_r_hand()
+		else
+			user.update_inv_l_hand()
+		return 1
+
+/obj/item/weapon/melee/arm_blade
+	name = "arm blade"
+	desc = "A grotesque blade made out of bone and flesh that cleaves through people as a hot knife through butter"
```
Comment (9600bauds) [inline code/game/gamemodes/changeling/changeling_powers.dm]: needs period reeee

**Comment by icantthinkofanameritenow** [comment]
As it turns out, porting shit is much harder than it looks. I'm still hoping to add the armblade eventually, but it'll have to be in a different PR where I actually know what the fuck I'm doing. Or for someone else who knows the code better to do it, whichever comes first.

**Comment by 9600bauds** [comment]
Looks like you're going to have to look through TG's codebase (or the codebase you got this from) and see what the missing functions and variables do, yeah it's not very fun

**Comment by icantthinkofanameritenow** [comment]
@9600bauds

That suggests that I even know where they are or how to convert them into our equivalents, which I don't know how to do right now. I think I'll stick to more modest changes for the time being.

**Comment by ihadtoregisterforthis** [comment]
Certain coder made it very clear that there is no NODROP on this codebase in #6198

Also 13commit

**Comment by icantthinkofanameritenow** [comment]
@ihadtoregisterforthis 
Yes, I know. That was mostly me trying to fix errors without using a dedicated Git client.

**Comment by clusterfack** [comment]
@icantthinkofanameritenow 
http://puu.sh/n7lT9/5de3989811.png

**Comment by icantthinkofanameritenow** [comment]
Well, as it turns out I don't have enough time.to code the armblades and all that anymore due to RL shit, but hopefully these price changes will still go through. The horror form cost reduction alone should ensure that there's going to be less monkeyman exploitation than before.

**Comment by ghost** [comment]
To be honest changelings need reworked over all but as a temporary fix this will be adequate :+1:

**Comment by gbasood** [comment]
Looks good to me. :+1: 
Since it's a pretty big balance change I'll leave this to @clusterfack or someone else more qualified than me
(you know, even though I merge big balance shit all the time)

**Comment by clusterfack** [comment]
These changes are fine just squash all these commits I guess and rewrite your OP to reflect whats in the PR

**Comment by 9600bauds** [comment]
and perform the making of the changelog

**Comment by Raumfahrer** [comment]
Can we change it so revives don't cost any chems?
It's fucking retarded that the easiest way to permanently kill a changeling isn't gib or incinerate the body, but to make it fight you using changeling powers then have somebody pop out of a closet and blow its brains out so it doesn't have enough chems to regen.

It's not fun, makes playing changeling shit and forces them all to resort to becoming maint-crawling powergame babbies because they go off like a glass cannon now.

**Comment by 9600bauds** [comment]
C'mon, fix this up before it conflicts

**Comment by gbasood** [comment]
squash + changelog and we're good to go

**Comment by icantthinkofanameritenow** [comment]
I added the changelog, but for ehatever reason this damn thing simply WILL NOT SQUASH. 

Addendum: Got it squashed after too fucking long.

**Pull request merged**